core/src/main/scala/io/projectglow/vcf/VariantContextToInternalRowConverter.scala

```diff
@@ -547,16 +548,17 @@ class VariantContextToInternalRowConverter(
             )
             formatKeysParsedWithoutHeader.add(key)
           }
-          val valueStr = VariantContextToVCFRowConverter.parseObjectAsString(
-            obj2any(identity)(genotype.getExtendedAttribute(key))
-          )
-          if (valueStr.nonEmpty) {
-            keys.append(UTF8String.fromString(key))
-            values.append(UTF8String.fromString(valueStr))
-          } else {
-            provideWarning(
-              s"Key $key has empty value $valueStr, but FLAG is not supported in FORMAT fields."
-            )
+          val value = genotype.getExtendedAttribute(key)
+          value match {
+            case _: JBoolean =>
+              provideWarning(
+                s"Key $key has an empty value, but FLAG is not supported in FORMAT fields."
+              )
```

Comment:
Why is this warning associated with JBoolean type?

Reply:
FORMAT fields should not have a FLAG type (boolean value); I'll change the message to make it clearer.